drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -605,23 +687,20 @@ static int stm32_i2c_msg_write(const struct device *dev, struct i2c_msg *msg,
 	stm32_i2c_enable_transfer_interrupts(dev);
 	LL_I2C_EnableIT_TX(i2c);
 
-	if (k_sem_take(&data->device_sync_sem,
-		       K_MSEC(STM32_I2C_TRANSFER_TIMEOUT_MSEC)) != 0) {
+	if (k_sem_take(&data->device_sync_sem, K_MSEC(STM32_I2C_TRANSFER_TIMEOUT_MSEC)) != 0) {
 		stm32_i2c_master_mode_end(dev);
 		k_sem_take(&data->device_sync_sem, K_FOREVER);
 		is_timeout = true;
 	}
 
```

Comment:
For devices with DCACHE enabled and rx buffer is in cached mem, it think it cache invalidation should be done here
Another option would be to issue a log warning when transfer is started that memory should be non-cacheable like it is done in uart-stm32.c

Reply:
I've added cache support, slightly modified from the UART implementation see if it makes sense where I put it. I don't have a platform at hand to test it though. I'll have to dig through my box of bits and see if I have something